traceevents/src/main/java/com/tomtom/kotlin/traceevents/TraceEvent.kt

```diff
@@ -42,8 +44,19 @@ data class TraceEvent(
     val interfaceName: String,
     val stackTraceHolder: Throwable?,
     val eventName: String,
-    val args: Array<Any?>
+    val args: Array<Any?>,
+    val parameterNames: Array<String>
 ) {
+    /**
+     * Get the method parameters as a map that maps the parameter name to its value.
+     */
+    fun getParametersMap(): Map<String, Any?> {
+        assert(args.size == parameterNames.size)
+        var map: MutableMap<String, Any?> = mutableMapOf()
+        args.indices.forEach { map.put(parameterNames[it], args[it]) }
```

Comment:
Instead of manually creating a map and populating it with `forEach`, you can use one of Kotlin's `associate...` functions. E.g., https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/associate-by.html